lib/adapter.js

```diff
@@ -146,3 +154,17 @@ adapterFactory.run = async function (cid, config, specs, capabilities) {
 
 export default adapterFactory
 export { CucumberAdapter, adapterFactory }
+
+function isAsync (func) {
+    const string = func.toString().trim()
+    return !!(
+        // native
+        string.match(/^async /) ||
+        // babel (this may change, but hey...)
+        string.match(/return _ref[^.]*\.apply/) ||
```

Comment:
Not sure if we really want to do that. I think it is sophisticated enough when we check if the function name is `async` or the config is `sync: false`. If the user wants to use async/await we should expect him to turn sync off in the config.